addons/ui/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/internal/util/SseUtil.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.ui.cometvisu.internal.util;
+
+import javax.servlet.ServletRequest;
+import javax.ws.rs.core.MediaType;
+
+import org.glassfish.jersey.media.sse.OutboundEvent;
+import org.openhab.ui.cometvisu.backend.beans.StateBean;
+import org.openhab.ui.cometvisu.internal.StateBeanMessageBodyWriter;
+
+/**
+ * Utility class containing helper methods for the SSE implementation.
+ * 
+ * @author Tobias Bräutigam - Initial Contribution and API
+ * 
+ */
+public class SseUtil {
+
+    static {
+        boolean servlet3 = false;
+        try {
+            servlet3 = ServletRequest.class.getMethod("startAsync") != null;
+        } catch (Exception e) {
+        } finally {
+            SERVLET3_SUPPORT = servlet3;
+        }
+    }
+
+    /**
+     * True if the {@link ServletRequest} class has a "startAsync" method,
+     * otherwise false.
+     */
+    public static final boolean SERVLET3_SUPPORT;
```

Comment:
If it makes your code easier: You can safely assume that openHAB 2 will ALWAYS have servlet3 support; the code was only introduced in Eclipse SmartHome since some other solutions besides openHAB might still use servlet 2.x.